app/src/main/java/com/example/studentarena/fragments/ProfileFragment.java

```diff
@@ -100,6 +102,20 @@ public void onClick(View v) {
         rvPosts.setAdapter(adapter);
         rvPosts.setLayoutManager(new GridLayoutManager(getContext(), 2));
         queryPosts();
+
+        swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+        // Setup refresh listener which triggers new data loading
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                queryPosts();
+            }
+        });
+        // Configure the refreshing colors
+        swipeContainer.setColorSchemeResources(android.R.color.holo_blue_bright,
+                android.R.color.holo_green_light,
+                android.R.color.holo_orange_light,
+                android.R.color.holo_red_light);
```

Comment:
`FeedFragment`, `ProfileFragment`, and probably however you implement search will have a few dozen lines of very very similar code to fetch posts and set up this refresh gesture. when it's time to polish your code later, i think we can figure out a way to write those things a single time and share it between the different usages. maybe `LayoutInflator` would let us define the feed and SwipeRefreshLayout views a single time in XML?